src/main/kotlin/no/nav/syfo/melding/MeldingService.kt

```diff
@@ -21,10 +26,22 @@ class MeldingService(
         // TODO: legg til i joark-cronjob og send på kafka her
     }
 
-    fun getMeldingerTilBehandler(personIdent: PersonIdent): List<MeldingTilBehandler> {
-        // TODO: Her ville vi vel gruppere det på en eller annen måte for ulike samtaler/dialoger
-        return database.getMeldingerForArbeidstaker(personIdent)
-            .filter { !it.innkommende } // Foreløpig bare ta MeldingTilBehandler-meldingene
-            .map { it.toMeldingTilBehandler() }
+    fun getConversationMeldingerMap(personIdent: PersonIdent): Map<UUID, List<Melding>> {
+        val meldinger = database.getMeldingerForArbeidstaker(personIdent)
+        return meldinger.groupBy(
+            keySelector = { it.conversationRef },
+            valueTransform = {
+                if (it.innkommende) {
+                    val behandlerRef: UUID = database.getMeldingerForConversation(it.conversationRef)
```

Comment:
Kanskje refaktorere ut dette til en metode i servicen for å finne `behandlerRef` gitt `conversationRef`